hydra/_internal/grammar/grammar_functions.py

```diff
@@ -309,7 +309,7 @@ def _sort_sweep(
     sweep = copy(sweep)
 
     if isinstance(sweep, ChoiceSweep):
-        sweep.list = sorted(sweep.list, reverse=reverse)
+        sweep.list = sorted(sweep.list, reverse=reverse)  # type: ignore
```

Comment:
I tried a couple of things like the following but still get the same mypy error, so had to go with ignore. 
```
sweep_list = sweep.list
assert isinstance(sweep_list, List)
sweep.list = sorted(sweep_list, reverse=reverse)

```